### PR TITLE
Change title check to match on TwitchSongRequests

### LINKED ISSUE #79: Update logic to expect the channel point reward to include TwitchSongRequests in order to activate
There are other things a broadcaster could want "song request" for, so open up that namespace by expecting `TwitchSongRequests` somewhere in the title or description of the reward. This might require some research in order to see what the best user experience would be for this.

### PR DESCRIPTION
**Pull request by agsaghhk**
Resolves #79

### CODE REVIEW

File: server/src/test/scala/com/github/saxypandabear/songrequests/websocket/TwitchSocketSpec.scala [hunk 1]
```diff
@@ -41,15 +41,18 @@ class TwitchSocketSpec extends UnitSpec {
...
+    socket.isSongRequest("This is a TwitchSongRequests reward") should be(true)
+    socket.isSongRequest("TwitchSongRequests") should be(true)
   }
 
   "Checking a reward title that does not contain the exact phrase" should "return false" in {
     socket.isSongRequest("This has a song, but is not a request") should be(
         false
     )
+    socket.isSongRequest("twitchsongrequests") should be(false)
+    socket.isSongRequest("This is a song request reward") should be(true)
```
Comment (SaxyPandaBear) [inline server/src/test/scala/com/github/saxypandabear/songrequests/websocket/TwitchSocketSpec.scala:54]: These tests are copy pasted wrong, and also don't actually test the thing it should be

File: server/src/main/scala/com/github/saxypandabear/songrequests/websocket/TwitchSocket.scala [hunk 2]
```diff
@@ -162,12 +163,13 @@ class TwitchSocket(
...
-   * Simple check on the reward title to make sure that it contains the phrase "song request"
+   * Simple check that the reward title is "TwitchSongRequests"
    * @param rewardTitle Title of the channel point redemption
-   * @return true if the title contains "song request", false otherwise
+   * @return true if the title is "TwitchSongRequests", false otherwise
    */
   private[websocket] def isSongRequest(rewardTitle: String): Boolean =
-    rewardTitle != null && rewardTitle.toLowerCase().contains("song request")
+    rewardTitle != null && rewardTitle
+      .matches(ExpectedRewardTitle)
```
Comment (SaxyPandaBear) [inline server/src/main/scala/com/github/saxypandabear/songrequests/websocket/TwitchSocket.scala:172]: Already talked about this, but it should be a contains check

File: server/src/main/scala/com/github/saxypandabear/songrequests/websocket/TwitchSocket.scala [hunk 3]
```diff
@@ -31,9 +31,10 @@ class TwitchSocket(
     pingFrequencyMs: Long = 60000
 ) extends LazyLogging {
 
-  private val spotifyUriPattern = "^(spotify:track:(\\w|\\d)+)$".r
-  private val pingTask          = new Timer()
-  private var session: Session  = _
+  private val ExpectedRewardTitle = "TwitchSongRequests"
```
Comment (SaxyPandaBear) [inline server/src/main/scala/com/github/saxypandabear/songrequests/websocket/TwitchSocket.scala:34]: You know, I'm not all that picky, but I think for constants I should be doing capitalized variable names like what you did here. 🤔 but I didn't. Maybe I'll go through and update those everywhere some other time. I think I have done all caps and snake case

Comment (agsaghhk) [reply server/src/main/scala/com/github/saxypandabear/songrequests/websocket/TwitchSocket.scala:34]: I wasn't sure what scala style was so I just referenced [this guide](https://docs.scala-lang.org/style/naming-conventions.html#constants-values-variable-and-methods). I wasn't sure if the codebase overall had a style choice for constants

Comment (SaxyPandaBear) [reply server/src/main/scala/com/github/saxypandabear/songrequests/websocket/TwitchSocket.scala:34]: We can come up with one but not required at all for this. I'm fine leaving this, updating other constants, and then defining that in a style guide

**Review by SaxyPandaBear — APPROVED** [review]

**Pull request merged**